packages/workbox-build/src/lib/errors.js

```diff
@@ -104,4 +104,10 @@ module.exports = {
   'bad-manifest-transforms-return-value': ol`The return value from a
     manifestTransform should be an object with 'manifest' and optionally
     'warnings' properties.`,
+  'string-entry-warning': ol`Some URLs were passed to additionalManifestEntries
+    as strings. Please ensure that these URLs contain versioning info
+    (e.g. 'https://example.com/v1.0/index.js'), as Workbox cannot maintain
+    its own revision info for them. To disable this warning, pass in an
```

Comment:
I'm not sure what you mean. Our build tools can't maintain revision info for those URLs, because they're (pretty much by definition) URLs that exist outside of the local build process.

If a developer has their own approach for revisioning these additional URLs then they're free to pass in meaningful info for the `revision` property instead of `null`. But in terms of this warning, I'm assuming that if they passed in a `string` it's because they don't have out-of-band revision info available.

Reply:
We could make an HTTP request, hash the contents of the response, and then use that as the revision (just as we do with files on the file system).

Then, if the file changes on the external server, a rebuild with the same user input would invalidate the old URL.

I definitely don't think this should be the default behavior; I'm just thinking it might potentially be a nice feature for developers who don't want/know how to do that themselves.

Reply:
I'm afraid we're opening up a can of worms here, because I *really* don't want developers to use this with unversioned URLs in the way you describe.

If there's a separate deployment process in place for those additional URLs, and it's not tied to generating the service worker, then developers are inevitably going to get confused when the additional URL is updated and their users don't see any updates until they remember to re-run their build process.

Reply:
Sure, just wanted to make sure you'd considered it.

Reply:
is using webpack's `[hash]` sufficient enough?
```js
        additionalManifestEntries: [
          { url: '/offline.html', revision: '[hash]'}
        ]

```

Reply:
If you have an actual hash value for a given asset derived from the current `webpack` compilation in a variable, it would be. But the syntax you're using there won't work; `workbox-webpack-plugin` won't automatically replace `'[hash]'` inside of `additionalManifestEntries` for you.

Reply:
😅 you're right! 
I ended up not needing `additionalManifestEntries` (for now at least)